### PR TITLE
Add Tests for Card Template Editor BottomNavigationView 

### PR DESCRIPTION
**Pull request by Akshay0701**
## 1 Fixed Bug : Show Updated Content In EditText(CardTemplateEditor)

Steps to reproduce the bug 

- open the card template editor 
- edit content Front or back.
- change to style 
- again change to front or back

Expected Output 
Edittext should show updated content 

Actual Output 
Editext showing previous version of the content 

## 2 Added Tests For BottomNavigationView

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplateEditorTest.java [hunk 1]
```diff
@@ -536,6 +536,69 @@ public void testDeckOverride() {
...
+
+        // set Bottom Navigation View to Front
+        cardTemplateFragment.setCurrentEditorView(R.id.front_edit, tempModel.getTemplate(0).getString("qfmt"), R.string.card_template_editor_front);
+
+        // check if current content is updated or not
+        assumeThat(templateEditText.getText().toString(), is(updatedFrontContent));
+    }
+
+    @Test
+    public void testCheckBottomNavigationView() {
```
Comment (Akshay0701) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplateEditorTest.java:570]: should I change this name to `testBottomNavigationView`

Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/CardTemplateEditorTest.java:570]: `testBottomNavigationViewLayoutTransition` maybe?

File: AnkiDroid/src/test/java/com/ichi2/anki/CardTemplateEditorTest.java [hunk 2]
```diff
@@ -536,6 +536,69 @@ public void testDeckOverride() {
 
     }
 
+    @Test
+    public void testContentPreserveAfterChangingEditorView() {
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/CardTemplateEditorTest.java:540]:
```suggestion
    public void testContentPreservedAfterChangingEditorView() {
```

**Review by david-allison-1 — APPROVED** [review]
LGTM, minor naming suggestions

**Comment by Akshay0701** [comment]
Thanks!

**Review by mikehardy — APPROVED** [review]
A big thank you for 1) double-checking your work over and over even after merge - this is how to really produce high quality work over long periods of time and 2) adding tests for anything eventually discovered.

That combination gives me confidence as I merge things for alphas - there may be problems initially but over time they will be corrected (and corrected forever, via tests detecting future problems)

**Pull request merged**